Engine/DataFeeds/LiveTradingDataFeed.cs

```diff
@@ -114,116 +106,40 @@ public bool IsActive
+        /// Creates a new subscription to provide data for the specified security.
         /// </summary>
         /// <param name="request">Defines the subscription to be added, including start/end times the universe and security</param>
-        /// <returns>True if the subscription was created and added successfully, false otherwise</returns>
-        public bool AddSubscription(SubscriptionRequest request)
+        /// <returns>The created <see cref="Subscription"/> if successful, null otherwise</returns>
+        public Subscription CreateSubscription(SubscriptionRequest request)
         {
-            if (_subscriptions.Contains(request.Configuration))
-            {
-                // duplicate subscription request
-                return false;
-            }
-
             // create and add the subscription to our collection
             var subscription = request.IsUniverseSubscription
                 ? CreateUniverseSubscription(request)
-                : CreateSubscription(request);
+                : InternalCreateSubscription(request);
 
-            // for some reason we couldn't create the subscription
-            if (subscription == null)
+            // check if we could create the subscription
+            if (subscription != null)
             {
-                Log.Trace("Unable to add subscription for: " + request.Configuration);
-                return false;
-            }
-
-            Log.Trace("LiveTradingDataFeed.AddSubscription(): Added " + request.Configuration);
-
-            _subscriptions.TryAdd(subscription);
-            // send the subscription for the new symbol through to the data queuehandler
-            // unless it is custom data, custom data is retrieved using the same as backtest
-            if (!subscription.Configuration.IsCustomData)
-            {
-                _dataQueueHandler.Subscribe(_job, new[] {request.Security.Symbol});
+                // send the subscription for the new symbol through to the data queuehandler
+                // unless it is custom data, custom data is retrieved using the same as backtest
+                if (!subscription.Configuration.IsCustomData)
+                {
+                    _dataQueueHandler.Subscribe(_job, new[] { request.Security.Symbol });
+                }
             }
 
-            return true;
+            return subscription;
         }
 
         /// <summary>
         /// Removes the subscription from the data feed, if it exists
         /// </summary>
-        /// <param name="configuration">The configuration of the subscription to remove</param>
-        /// <returns>True if the subscription was successfully removed, false otherwise</returns>
-        public bool RemoveSubscription(SubscriptionDataConfig configuration)
+        /// <param name="subscription">The subscription to remove</param>
+        public void RemoveSubscription(Subscription subscription)
```

Comment:
Interesting to note that the removal semantics are different for backtesting and live -- no need to change as part of this PR (or maybe at all), but worth thinking of ideas for how a subscription can clean himself up -- perhaps live subscriptions implement a `Dispose` method that performs the behavior required here in `RemoveSubscription` -- this would end up (methinks) with removing `IDataFeed.RemoveSubscription`

Reply:
Good idea! That would make it look even cleaner